tools/mpremote/mpremote/main.py

```diff
@@ -355,17 +357,23 @@ def load_user_config():
     # Create empty config object.
     config = __build_class__(lambda: None, "Config")()
     config.commands = {}
-
-    # Get config file name.
-    path = os.getenv("XDG_CONFIG_HOME")
-    if path is None:
-        path = os.getenv("HOME")
-        if path is None:
-            return config
-        path = os.path.join(path, ".config")
-    path = os.path.join(path, _PROG)
+    # use $XDG_CONFIG_HOME,$HOME $env:USERPROFILE% or $env:APPDATA
+    path = None
+    for env_var in ("XDG_CONFIG_HOME", "HOME", "USERPROFILE", "APPDATA"):
+        path = os.getenv(env_var)
+        if not path:
+            continue
+        if os.path.exists(os.path.join(path, ".config", _PROG, "config.py")):
+            # Unix style
+            path = os.path.join(path, ".config", _PROG, "config.py")
+            break
+        elif os.path.exists(os.path.join(path, _PROG, "config.py")):
+            # Windows style
```

Comment:
Not sure it's worth making this distinction, there's plenty of programs (including micropython itself) which just put dot directories in in the home directory on windows as well.

Reply:
Agree that adding an additional layer of .config does not help, but that is the **current** behavior of mpremote, and removing that would result in a breaking change for all current *nix users that have a config.
That is fine with me - but probably something that would need to be called out in the change notes. 
The Freedektop specs are not that clear either to me 
https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html
but that is the most likely source for the `.config` that I could pinpoint: "If $XDG_CONFIG_HOME is either not set or empty, a default equal to $HOME/.config should be used."

Also, based on your earlier feedback , there may be Windows users that use MSYS2/mingw, and expect to use a Unix convention on Windows, hence the check for Unix Style on Windows.

Reply:
Perhaps @jimmo or @dpgeorge can help resolve the intended paths to use across the different client platforms.
I also found [a similar conversions on SO](https://stackoverflow.com/a/22161688/7909427) regarding which platforms use what paths for configuration.

currently my recommendation would be:

| # | ENV_VAR         | env_value                    | Path_checked                              | platform.system()              |  |
| - | --------------- | ---------------------------- | ----------------------------------------- | ------------------------------ | - |
| 1 | XDG_CONFIG_HOME | /home/foo/.config            | /home/foo/.config/mpremote                | Linux, Windows(MSYS32), Darwin |  |
| 2 | HOME            | /home/boo                    | /home/foo/.config/mpremote                | Linux, Windows(MSYS32), Darwin |  |
| 3 | USERPROFILE     | C:\Users\foo                 | C:\Users\foo\mpremote                     | Windows                        |  |
| 4 | APPDATA         | C:\Users\foo\AppData\Roaming | C:\Users\josverl\AppData\Roaming\mpremote | Windows                        |  |

That means that `.config` only will be added to `$HOME`, but please validate my understanding.
 
 Once we agree on what the outcome should be, I'll update the code accordingly

Reply:
> and removing that would result in a breaking change for all current *nix users that have a config

Sorry should have been more clear, I'm not saying the current behavior should change, just that as far as I'm concerned it's not needed to that extra location (`os.path.join(path, _PROG, "config.py")`) just for windows, but instead have it also use the same as *nix (`os.path.join(path, ".config", _PROG, "config.py")`)

Reply:
It just does not make sense to enforce Linux naming conventions to Windows userprofiles.
I know I may be opiniated as one of these OSes is created by my current employer, but I stand by this design.